src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.ReadHelper.cs

```diff
@@ -29,11 +29,11 @@ public partial class JsonTypeInfo<T>
             ReadStack readStack = default;
             readStack.Initialize(this, supportContinuation: true);
             var jsonReaderState = new JsonReaderState(options.GetReaderOptions());
-            ReadBufferState bufferState = null;
+            // bufferState is initialized last to exlcude any exceptions before entering the try/finally block where bufferState.Dispose call is executed
+            var bufferState = new ReadBufferState(options.DefaultBufferSize);
 
             try
```

Comment:
A terser message and tactical use of indentation should suffice to convey the message:
```suggestion

            // Note: ReadBufferState rents pooled buffers.
            ReadBufferState bufferState = new(options.DefaultBufferSize);
            try
```

Reply:
Fixed